src/elements/filter-menu.html

```diff
@@ -0,0 +1,235 @@
+            filter-key$="[[selectedFilter.key]]"
+            filter-value$="[[selectedFilter.value]]"
+            on-tap="_toggleFilter"
+            selected
+            layout horizontal inline center
+          >
+            <span>[[selectedFilter.value]]</span>
+            <iron-icon icon="hoverboard:close"></iron-icon>
+          </div>
+        </template>
+      </div>
+    </div>
+
+    <div class="filters-board" block$="[[_openedBoard]]">
+      <div class="container">
+        <template is="dom-repeat" items="[[filters]]" as="filter">
+          <div class="filter-group">
+            <h3 class="filter-title">[[filter.title]]</h3>
+            <template is="dom-repeat" items="[[filter.items]]" as="item">
+              <div
+                class="tag"
+                style$="--color: [[getVariableColor(item, 'primary-text-color')]]"
+                filter-key$="[[filter.key]]"
+                filter-value$="[[item]]"
+                selected$="[[_isSelected(selected, filter.key, item)]]"
+                on-tap="_toggleFilter"
+              >
+                [[item]]
+              </div>
+            </template>
+          </div>
+        </template>
+      </div>
+    </div>
+
+
+  </template>
+
+  <script>
+
+    // eslint-disable-next-line no-undef
+    class FilterMenu extends UtilsFunctions(Polymer.Element) {
+      static get is() {
+        return 'filter-menu';
+      }
+
+      static get properties() {
+        return {
+          filters: Array,
+          resultsCount: Number,
+          selected: {
+            type: Object,
+            value: {},
+          },
+          _selectedArray: {
+            type: Array,
+            computed: '_generateSelectedArray(selected)',
+          },
+          _openedBoard: {
+            type: Boolean,
+            value: false,
+          },
+        };
+      }
+
+      _isSelected(selectedFilters, key, value) {
+        return selectedFilters[key]
+          && selectedFilters[key].includes(this.generateClassName(value.trim()));
+      }
+
+      _toggleFilter(e) {
+        const filterKey = e.currentTarget.getAttribute('filter-key');
+        const filter = this.generateClassName(e.currentTarget.getAttribute('filter-value').trim());
+        let resultQueryObject;
+
+        if (this.selected[filterKey]) {
+          resultQueryObject = Object.assign({}, this.selected, {
+            [filterKey]: this._isSelected(this.selected, filterKey, filter)
+              ? this.selected[filterKey].filter((selectedFilter) => selectedFilter !== filter)
+              : this.selected[filterKey].concat([filter]),
+          });
+        } else {
+          resultQueryObject = Object.assign({}, this.selected, {
+            [filterKey]: [filter],
+          });
+        }
+
+        const searchParams = Object.entries(resultQueryObject).reduce((aggr, entry) => {
+          const filters = entry[1].map((filter) => `${entry[0]}=${filter}`);
```

Comment:
Can it be more explainable, what does mean entry[1]? :)

Reply:
It contains value 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries